homeassistant/components/rachio/__init__.py

```diff
@@ -349,6 +348,23 @@ def stop_watering(self) -> None:
         _LOGGER.info("Stopped watering of all zones on %s", str(self))
 
 
+class RachioDeviceMixIn(Entity):
+    """Mixin to provide device_info."""
+
+    @property
```

Comment:
Please also define a common init method, that sets the self._controller instance attribute etc.

I also see that we should move the callback registration with `dispatcher_connect` from the init methods of the entities to the entity coroutine method `async_added_to_hass`.

Reply:
done